controllers/operands/deploymentHandler_test.go

```diff
@@ -0,0 +1,105 @@
+package operands
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/commontestutils"
+)
+
+var _ = Describe("Deployment Handler", func() {
+	Context("update or recreate the Deployment as required", func() {
+		var hco *hcov1beta1.HyperConverged
+		var req *common.HcoRequest
+		var originalDeployment *appsv1.Deployment
+		var modifiedDeployment *appsv1.Deployment
+
+		BeforeEach(func() {
+			hco = commontestutils.NewHco()
+			req = commontestutils.NewReq(hco)
+			originalDeployment = &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Deployment",
+					APIVersion: "apps/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "modifiedDeployment",
+					Labels: map[string]string{"key1": "value1"},
+					UID:    "testuid",
+				},
+				Spec: appsv1.DeploymentSpec{},
+			}
+			modifiedDeployment = &appsv1.Deployment{}
+			originalDeployment.DeepCopyInto(modifiedDeployment)
+		})
+
+		It("should recreate the Deployment as LabelSelector has changed", func() {
+			// create a fake client using original deployment
+			cl := commontestutils.InitClient([]client.Object{originalDeployment})
+			foundResource := &appsv1.Deployment{}
+			Expect(
+				cl.Get(context.TODO(),
+					types.NamespacedName{Namespace: originalDeployment.GetNamespace(), Name: originalDeployment.GetName()},
+					foundResource),
+			).ToNot(HaveOccurred())
+
+			// modify the LabelSelector
+			modifiedDeployment.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: map[string]string{"key2": "value2"},
+			}
+			modifiedDeployment.SetUID("")
```

Comment:
@tiraboschi I've set an empty UID in this test case because we want to be sure that the UID changes at the end.